vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/DashboardItemReorderEndEvent.java

```diff
@@ -34,19 +46,99 @@ public class DashboardItemReorderEndEvent extends ComponentEvent<Dashboard> {
      * @param fromClient
      *            <code>true</code> if the event originated from the client
      *            side, <code>false</code> otherwise
+     * @param items
+     *            The ordered items represented by node IDs as a
+     *            {@link JsonArray}
      */
     public DashboardItemReorderEndEvent(Dashboard source, boolean fromClient,
             @EventData("event.detail.items") JsonArray items) {
         super(source, fromClient);
-        this.items = items;
+        setReorderedItemParent(source, items);
+        if (reorderedItemsParent == null) {
+            // No reordering
+            reorderedItemsParent = source;
+            reorderedItems = source.getChildren().toList();
+        } else {
+            setReorderedItems();
+        }
+    }
+
+    /**
+     * Returns the parent of the reordered items. Either a dashboard or a
+     * section.
```

Comment:
I tried this locally and I'm always getting the `Dashboard` instance, even when reordering inside a section.